js/test/test.js

```diff
@@ -173,17 +200,27 @@ async function loadExchange (exchange) {
         'EUR/USD',
     ];
 
-    let result = exchange.symbols.filter ((symbol) => symbols.indexOf (symbol) >= 0);
+    const resultSymbols = [];
+    const exchangeSpecificSymbols = exchange.symbols;
+    for (let i = 0; i < exchangeSpecificSymbols.length; i++) {
+        const symbol = exchangeSpecificSymbols[i];
+        if (exchange.inArray(symbol, symbols)) {
+            resultSymbols.push (symbol);
+        }
+    }
 
-    if (result.length > 0) {
-        if (exchange.symbols.length > result.length) {
-            result = result.join (', ') + ' + more...';
+    let resultMsg = '';
+    const resultLength = resultSymbols.length;
+    const exchangeSymbolsLength = exchange.symbols.length;
+    if (resultLength > 0) {
+        if (exchangeSymbolsLength > resultLength) {
+            resultMsg = resultSymbols.join (', ') + ' + more...';
```

Comment:
we'd better use a new variable `resultMsg` instead of same `result` variable being rewriten with from `array` to `string` type.